test/nodetool/conftest.py

```diff
@@ -27,31 +27,52 @@ def pytest_addoption(parser):
                      " with --nodetool=cassandra, this should be the nodetool binary")
     parser.addoption('--jmx-path', action='store', default=None,
                      help="Path to the jmx binary, only used with --nodetool=cassandra")
+    parser.addoption('--run-within-unshare', action='store_true',
+                     help="Setup the 'lo' network if launched with unshare(1)")
 
 
 @pytest.fixture(scope="session")
-def rest_api_mock_server():
-    ip = f"127.{random.randint(0, 255)}.{random.randint(0, 255)}.{random.randint(0, 255)}"
-    port = random.randint(10000, 65535)
+def maybe_setup_loopback_network(request):
```

Comment:
Just `loopback_network` is a better enough name here. With fixtures, all we care about is the end result. At the end of this fixture we will have a `lo` interface setup, either because we did set it up, or because it was already set up.

BTW, can we move this logic to `test.py`, so all tool suites already have a functioning `lo` when launched (I also suggested we launch with `unshare` unconditionally.

Reply:
also, we need to use a "driver" to create the network namespace, and teach the test to enter the namespace. the `test.py` is complicated enough. i don't really want to make it a "i will take care of this and, yes that also" kind of baby sitter. i think, test suite should take care of themselves as long as they can.

Reply:
the reason I proposed this is because I have another PR (maybe I closed) which adds another tool suite, and it will also need `unshare`, just like the `nodetool` one. Right now this doesn't make a difference, but with two or more `tool` suites, pushing this to `test.py` allows for deduplication.

We can do this later, the current code is good enough as-is.

Reply:
once we have more needs in this regard, probably we could promote this fixture into `test/pylib` ?

Reply:
It doesn't work, fixtures have to be declared in `conftest.py`, otherwise `pytest` won't pick them up.

Reply:
yeah, i see. i meant, we could have a tiny class to encapsulate the related logic: the option, and the lo interface. but they are very short and straightforward. maybe we can just replicate them..

Reply:
I think just moving this to `test.py` will be even simpler. But we'll see when (if) we get there.